wrangler-docs/functions/logical.md

```diff
@@ -0,0 +1,161 @@
+
+#### Output
+long
+
+#### Example
+If `a` contains the number `352` and `b` contains the number `400`,
+then the following two functions are equivalent, and return the value `496`.
+
+```
+set-column val logical:BitOr(352,400)
+set-column val logical:BitOr(a,b)
+```
+
+## BitXor
+Returns the bitwise `XOR` of the two arguments. The inputs must be non-null.
+
+#### Namespace
+`logical`
+
+#### Input
+number1(long, int), number2(long, int)
+
+#### Output
+long
+
+#### Example
+
+If `a` contains the number `352` and `b` contains the number `400`,
+then the following two functions are equivalent, and return the value `240`:
+
+```
+set-column val logical:BitXOr(352,400)
+set-column val logical:BitXOr(a,b)
+```
+
+## BitCompress
+Returns the `integer` made from the string argument,
+which contains a binary representation of `"1"s` and `"0"s`.
+
+#### Namespace
+`logical`
+
+#### Input
+String
+
+#### Output
+number(long)
+
+#### Example
+If `mynumber` contains the string `"101100000"`,
+then the following two functions are equivalent, and return the number `352`.
+```
+set-column val logical:BitCompress("101100000")
+set-column val logical:BitCompress(mynumber)
+```
+
+## BitExpand
+Returns a `string` containing the binary representation in `"1"s` and `"0"s`
+of the given `long`.
+
+#### Namespace
+`logical`
+
+#### Input
+number(long)
+
+#### Output
+String
+
+#### Example
+If `mynumber` contains the number `352`, then the following two functions
+are equivalent, and return the string `"101100000"`
+```
+set-column val logical:BitExpand(352)
+set-column val logical:BitExpand(mynumber)
+```
+
+## Not
+Returns the Not of the logical value of an expression. If the value of expression is `true`, the Not function
+returns a value of `false (0)`. If the value of expression is `false`, the `NOT` function returns a value of `true (1)`.
+A numeric expression that evaluates to `0` is a logical value of `false`. A numeric expression that evaluates to
+anything else, other than the `null` value, is a logical `true`. An empty string is logically `false`. All other string
+expressions, including strings that include an empty string, spaces, or the number `0` and spaces, are logically `true`.
+
+#### Namespace
+`logical`
+
+#### Input
+expression(double, float, int, long, String)
+
+#### Output
+number (int)
+
+#### Example
+If `mynumber` contains the number `352`, then the following two functions are equivalent,
+and return the string `"101100000"`
+```
+set-column val logical:Not(5-5)
+set-column val logical:Not(5+5)
+set-column val logical:Not('value')
+```
+
+## SetBit
+Returns an `long` with specific bits set to a specific state, where `'origValue'` is the input value to perform
+the action on, `'bitArray'` is a array containing a list bit numbers (starting from 1) to set the state of,
+and `'bitState'` is either `1` or `0`, indicating which state to set those bits.
+
+#### Namespace
+`logical`
+
+#### Input
+
+ - origValue (long)
```

Comment:
Here the inputs are on different lines, previously they are on the same line. Please make sure we use one consistent style.